internal/controller/stages/images.go

```diff
@@ -58,11 +61,29 @@ func (r *reconciler) getLatestImages(
 			)
 		}
 		imgs[i] = api.Image{
-			RepoURL: sub.RepoURL,
-			Tag:     tag,
+			RepoURL:    sub.RepoURL,
+			GitRepoURL: r.getImageSourceURL(sub.GitRepoURL, tag),
+			Tag:        tag,
 		}
 		logger.WithField("tag", tag).
 			Debug("found latest suitable image tag")
 	}
 	return imgs, nil
 }
+
+const (
+	githubURLPrefix = "https://github.com"
+)
+
+func (r *reconciler) getImageSourceURL(gitRepoURL, tag string) string {
+	for baseUrl, fn := range r.imageSourceURLFnsByBaseURL {
+		if strings.HasPrefix(gitRepoURL, baseUrl) {
+			return fn(gitRepoURL, tag)
+		}
+	}
+	return ""
+}
+
+func getGithubImageSourceURL(gitRepoURL, tag string) string {
+	return fmt.Sprintf("%s/tree/%s", git.NormalizeGitURL(gitRepoURL), tag)
+}
```

Comment:
I think this reconstruction can happen in UI? At least that's where Argo CD is doing it

Reply:
I think that's one way to do it, but this way, CLI and `kubectl` users still get the advantage of it.